src/ElasticApmAgentStartupHook/StartupHook.cs

```diff
@@ -104,7 +108,53 @@ public static void Initialize()
 			}
 		}
 
-		InvokerLoaderMethod(loader);
+		if (loaderAssembly is null)
+		{
+			_logger.WriteLine(
+				$"No {LoaderDll} assembly loaded. Agent not loaded");
+		}
+
+		LoadAssembliesFromLoaderDirectory(loaderDirectory);
+		InvokerLoaderMethod(loaderAssembly);
+	}
+
+	/// <summary>
+	/// Loads assemblies from the loader directory if they exist
+	/// </summary>
+	/// <param name="loaderDirectory"></param>
+	private static void LoadAssembliesFromLoaderDirectory(string loaderDirectory)
+	{
+		var context = new ElasticApmAssemblyLoadContext();
+		AssemblyLoadContext.Default.Resolving += (_, name) =>
```

Comment:
Previously when we had Newtonsoft.Json and Ben.Demystifier as normal dependencies, often I ran into stackoverflow exceptions with this. Seems working now, just noting it here.